lstchain/calib/camera/r0.py

```diff
@@ -346,6 +346,22 @@ def do_time_lapse_corr(waveform, expected_pixel_id, local_clock_list, fc, last_t
                         posads = int((k + fc[nr_module, gain, pix]) % size4drs)
                         last_time_array[nr_module, gain, pix, posads] = time_now
 
+                # now the magic of Dragon,
+                # if the ROI is in the last quarter of each DRS4
+                # for even channel numbers extra 12 slices are read in a different place
+                # code from Takayuki & Julian
+                if pix % 2 == 0:
+                    first_cap = fc[nr_module, gain, pix]
+                    if first_cap % 1024 > 766 and first_cap % 1024 < 1012:
+                        start = int(first_cap) + 1024 - 1
+                        end = int(first_cap) + 1024 + 11
+                        for kk in range(start, end):
```

Comment:
I think you may avoid the loop here with
`last_time_array[nr_module, gain, pix, start%4096:end%4096] = time_now`

Reply:
Done, thank you very much.